javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java

```diff
@@ -164,8 +165,14 @@ private static boolean isJar(File f) {
 
   private static void addFileUrl(List<URL> result, File file) {
     try {
-      URL wrappedUrl = new URL("otel", null, -1, "/", new RemappingUrlStreamHandler(file));
-      result.add(wrappedUrl);
+      // skip shading extension classes if opentelemetry-api is not shaded (happens when using
+      // disableShadowRelocate=true)
+      if (Context.class.getName().contains("shaded")) {
```

Comment:
(mostly for readability)

```suggestion
      if (Context.class.getName().contains(".shaded.")) {
```